### PR TITLE
gtk-vnc: update to 1.3.1, spice-gtk: update to 0.41

### PR DESCRIPTION
**Pull request by gasinvein**
#### Description

Update gtk-vnc to 1.3.1, dropping now-unneded patch.
Update spice-gtk to 0.41, dropping one now-unneded patch and rebasing remaining one.
This fixes quartz variants build at least on Monterey.
Both tested in virt-manager

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 12.5.1 21G83 arm64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Review by danchr — APPROVED** [review]
LGTM. Feel free to assume maintainership; I haven't used this port in ages.

**Comment by gasinvein** [comment]
I'm more or less familiar with MacPorts from a user perspective, but pretty new to developing/maintaining ports.

**Comment by mascguy** [comment]
> In order to build virt-manager quartz variant, updating `spice-gtk` is also needed. Should I open another PR for that, or pushing additional changes to this one is ok?

You're welcome to add that here, given that both are needed.

**Comment by gasinvein** [comment]
@mascguy Both done.

File: gnome/spice-gtk/Portfile [hunk 1]
```diff
@@ -5,11 +5,11 @@ PortGroup           active_variants 1.1
 PortGroup           meson 1.0
 
 name                spice-gtk
-version             0.39
+version             0.41
 revision            1
```
Comment (mascguy) [inline gnome/spice-gtk/Portfile:9]: When updating the version, you'll also want to reset `revision` to zero.

Comment (gasinvein) [reply gnome/spice-gtk/Portfile]: Thanks, done.

**Review by mascguy — APPROVED** [review]
CI passes, and your changes look good.

Thank you so much for your contribution!

**Pull request merged**